extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java

```diff
@@ -56,41 +56,43 @@ public ParquetFileReader(final URI parquetFileURI, final SeekableChannelsProvide
             // TODO(deephaven-core#5066): Add support for reading metadata files from non-file URIs
             rootURI = parquetFileURI;
         }
-        final byte[] footer;
-        try (final SeekableChannelContext channelContext = channelsProvider.makeContext();
-                final SeekableByteChannel readChannel =
-                        channelsProvider.getReadChannel(channelContext, parquetFileURI)) {
-            final long fileLen = readChannel.size();
-            if (fileLen < MAGIC.length + FOOTER_LENGTH_SIZE + MAGIC.length) { // MAGIC + data + footer +
-                // footerIndex + MAGIC
-                throw new InvalidParquetFileException(
-                        parquetFileURI + " is not a Parquet file (too small length: " + fileLen + ")");
+        try (
+                final SeekableChannelContext context = channelsProvider.makeSingleUseContext();
```

Comment:
We should probably use a regular context here, even though it feels like "single use".